studio/src/mas-repository.js

```diff
@@ -355,35 +401,89 @@ export class MasRepository extends LitElement {
         return fragment;
     }
 
-    /** Write */
-
     /**
-     * @returns {Promise<boolean>} Whether or not it was successful
+     * Unified method to save a fragment (regular or dictionary)
+     * @param {Object} fragment - The fragment to save
+     * @param {Object} options - Additional options
+     * @param {boolean} options.isInEditStore - Whether the fragment is from the inEdit store
+     * @returns {Promise<Object>} The saved fragment
      */
-    async saveFragment() {
-        Events.toast.emit({
-            variant: 'info',
-            content: 'Saving fragment...',
-        });
+    async saveFragment(fragment, options = {}) {
+        const { isInEditStore = true } = options;
+        const fragmentToSave = isInEditStore ? this.fragmentInEdit : fragment;
+        
+        if (!fragmentToSave) {
+            throw new Error('No fragment provided for saving');
+        }
+        
+        const isDictionaryFragment = fragmentToSave.path?.includes('/dictionary/');
+        this.showToast(isDictionaryFragment ? 'Saving placeholder...' : 'Saving fragment...');
```

Comment:
we shouldn't do that either, just saving frgment is fine

Reply:
done